src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -150,6 +150,16 @@ public function getServerParameter($key, $default = '')
         return isset($this->server[$key]) ? $this->server[$key] : $default;
     }
 
+    public function switchToXMLHttpRequest()
+    {
+        $this->setServerParameter('HTTP_X_REQUESTED_WITH', 'XMLHttpRequest');
+    }
+
+    public function removeXMLHttpRequest()
```

Comment:
I find this name a bit confusing. We are not removing any request :)

Maybe we can just only have a setter instead? `setXMLHttpRequest($isXMLHttpRequest)`

Reply:
I will add Header a the end I think